src/payment/strategies/ppsdk/ppsdk-strategy.ts

```diff
@@ -1,23 +1,50 @@
 import { CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
-import { OrderRequestBody } from '../../../order';
+import { NotInitializedError, NotInitializedErrorType } from '../../../common/error/errors';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
 import { PaymentInitializeOptions, PaymentRequestOptions } from '../../payment-request-options';
 import PaymentStrategy from '../payment-strategy';
 
+import { getPPSDKMethod } from './get-ppsdk-payment-method';
+import { getPaymentProcessor , PaymentProcessor } from './ppsdk-payment-processor';
+
 export class PPSDKStrategy implements PaymentStrategy {
+    private _paymentProcessor?: PaymentProcessor;
 
     constructor(
-        protected _store: CheckoutStore
+        private _store: CheckoutStore,
+        private _orderActionCreator: OrderActionCreator
     ) {}
 
-    execute(_payload: OrderRequestBody, _options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
-        return Promise.resolve(this._store.getState());
+    execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment, ...order } = payload;
+        const { _paymentProcessor: paymentProcessor } = this;
+
+        if (!paymentProcessor) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+
+        return this._store.dispatch(this._orderActionCreator.submitOrder(order, options))
+            .then(() => paymentProcessor(payment))
+            .then(() => this._store.getState());
     }
 
     finalize(_options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
         return Promise.resolve(this._store.getState());
     }
 
-    initialize(_options?: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+    initialize(options?: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+        const paymentMethod = getPPSDKMethod(this._store, options);
+
+        if (!paymentMethod) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+
+        this._paymentProcessor = getPaymentProcessor(paymentMethod);
```

Comment:
Prefer to inject dependencies to produce side effects. `getPaymentProcessor` is directly imported here so it's not possible to pass a different implementation in your tests besides using Jest mock. Generally speaking, we discourage the use of Jest mock because it requires you to peek into the implementation of your test subject in order to do the mocking; where as via constructor or method parameters, you can tell what to mock by looking at their interfaces.

Reply:
Done: https://github.com/bigcommerce/checkout-sdk-js/pull/1150/files#diff-d42088af7e8a0106b9a187cf956d965bc6d677dfa26f573607c994dbd3a75a62R395